nautobot/docs/development/navigation-menu.md

```diff
@@ -0,0 +1,129 @@
+# Navigation Menu Tabs
+
+Core applications and plugins can modify the existing navigation bar layout by defining `menu_items` inside of `navigation.py`. Using the key and weight system, a developer can integrate the plugin amongst existing menu tabs, groups, items and buttons and/or create entirely new menus as desired.
+
+## Adding a new tab
```

Comment:
I might suggest swapping the order of 'Adding a new tab' versus 'Modifying an existing tab', just so as to encourage the former where appropriate and reduce the proliferation of new menu tabs.